pandas/tests/frame/test_analytics.py

```diff
@@ -12,65 +12,58 @@
 from numpy.random import randn
 import numpy as np
 
-from pandas.compat import lrange, PY35
-from pandas import (compat, isna, notna, DataFrame, Series,
-                    MultiIndex, date_range, Timestamp, Categorical,
-                    _np_version_under1p12,
-                    to_datetime, to_timedelta)
-import pandas as pd
+from pandas.compat import lrange, PY3, PY35
+from pandas import (DataFrame, Series, MultiIndex, CategoricalIndex,
+                    DatetimeIndex, Categorical, Timestamp, Timedelta,
+                    concat, cut, isna, notna, date_range, timedelta_range,
+                    to_datetime, to_timedelta, NaT, option_context,
+                    _np_version_under1p12)
 import pandas.core.nanops as nanops
 import pandas.core.algorithms as algorithms
 
 import pandas.util.testing as tm
 import pandas.util._test_decorators as td
-from pandas.tests.frame.common import TestData
 
 
-class TestDataFrameAnalytics(TestData):
+class TestDataFrameAnalytics():
 
     # ---------------------------------------------------------------------=
     # Correlation and covariance
 
     @td.skip_if_no_scipy
-    def test_corr_pearson(self):
-        self.frame['A'][:5] = nan
-        self.frame['B'][5:10] = nan
+    def test_corr_pearson(self, float_frame):
+        float_frame['A'][:5] = nan
+        float_frame['B'][5:10] = nan
 
-        self._check_method('pearson')
+        self._check_method(float_frame, 'pearson')
 
     @td.skip_if_no_scipy
-    def test_corr_kendall(self):
-        self.frame['A'][:5] = nan
-        self.frame['B'][5:10] = nan
+    def test_corr_kendall(self, float_frame):
+        float_frame['A'][:5] = nan
+        float_frame['B'][5:10] = nan
 
-        self._check_method('kendall')
+        self._check_method(float_frame, 'kendall')
 
     @td.skip_if_no_scipy
-    def test_corr_spearman(self):
-        self.frame['A'][:5] = nan
-        self.frame['B'][5:10] = nan
+    def test_corr_spearman(self, float_frame):
+        float_frame['A'][:5] = nan
+        float_frame['B'][5:10] = nan
 
-        self._check_method('spearman')
+        self._check_method(float_frame, 'spearman')
 
-    def _check_method(self, method='pearson', check_minp=False):
-        if not check_minp:
-            correls = self.frame.corr(method=method)
-            exp = self.frame['A'].corr(self.frame['C'], method=method)
-            tm.assert_almost_equal(correls['A']['C'], exp)
-        else:
-            result = self.frame.corr(min_periods=len(self.frame) - 8)
-            expected = self.frame.corr()
-            expected.loc['A', 'B'] = expected.loc['B', 'A'] = nan
-            tm.assert_frame_equal(result, expected)
+    def _check_method(self, frame, method='pearson'):
```

Comment:
I had a typo in the else-branch of `check_minp`, and all tests still ran through, so I'm suggesting to remove this unused branch.